JSDemos/Demos/DropDownBox/SingleSelection/React/App.js

```diff
@@ -10,180 +10,144 @@ import 'whatwg-fetch';
+
+const treeDataSource = makeAsyncDataSource('treeProducts.json');
+const gridDataSource = makeAsyncDataSource('customers.json');
+
+function App() {
+  const treeViewRef = useRef(null);
+  const [treeBoxValue, setTreeBoxValue] = useState('1_1');
+  const [gridBoxValue, setGridBoxValue] = useState([3]);
+  const [isGridBoxOpened, setIsGridBoxOpened] = useState(false);
+  const [isTreeBoxOpened, setIsTreeBoxOpened] = useState(false);
+
+  const treeViewRender = useCallback(() => (
+    <TreeView dataSource={treeDataSource}
+      ref={treeViewRef}
+      dataStructure="plain"
+      keyExpr="ID"
+      parentIdExpr="categoryId"
+      selectionMode="single"
+      displayExpr="name"
+      selectByClick={true}
+      onContentReady={treeViewOnContentReady}
+      onItemClick={onTreeItemClick}
+      onItemSelectionChanged={treeViewItemSelectionChanged}
+    />
+  ), [treeDataSource]);
+
+  const dataGridRender = useCallback(() => (
+    <DataGrid
+      dataSource={gridDataSource}
+      columns={gridColumns}
+      hoverStateEnabled={true}
+      selectedRowKeys={gridBoxValue}
+      onSelectionChanged={dataGridOnSelectionChanged}
+      height="100%">
+      <Selection mode="single" />
+      <Scrolling mode="virtual" />
+      <Paging enabled={true} pageSize={10} />
+      <FilterRow visible={true} />
+    </DataGrid>
+  ), [gridDataSource, gridBoxValue]);
+
+  const syncTreeViewSelection = useCallback((e) => {
+    setTreeBoxValue(e.value);
+    if (!treeViewRef.current) return;
 
     if (!e.value) {
-      this.treeView.instance.unselectAll();
+      treeViewRef.current.instance.unselectAll();
     } else {
-      this.treeView.instance.selectItem(e.value);
+      treeViewRef.current.instance.selectItem(e.value);
     }
-  }
-
-  syncDataGridSelection(e) {
-    this.setState({
-      gridBoxValue: e.value,
-    });
-  }
-
-  treeViewItemSelectionChanged(e) {
-    this.setState({
-      treeBoxValue: e.component.getSelectedNodeKeys(),
-    });
-  }
-
-  dataGridOnSelectionChanged(e) {
-    this.setState({
-      gridBoxValue: e.selectedRowKeys,
-      isGridBoxOpened: false,
-    });
-  }
-
-  gridBoxDisplayExpr(item) {
-    return item && `${item.CompanyName} <${item.Phone}>`;
-  }
-
-  treeViewOnContentReady(e) {
-    e.component.selectItem(this.state.treeBoxValue);
-  }
-
-  onTreeItemClick() {
-    this.setState({
-      isTreeBoxOpened: false,
-    });
-  }
-
-  onGridBoxOpened(e) {
+  }, []);
+
+  const syncDataGridSelection = useCallback((e) => {
+    setGridBoxValue(e.value);
+  }, []);
+
+  const treeViewItemSelectionChanged = useCallback((e) => {
+    setTreeBoxValue(e.component.getSelectedNodeKeys());
+  }, []);
+
+  const dataGridOnSelectionChanged = useCallback((e) => {
+    setGridBoxValue(e.selectedRowKeys);
+    setIsGridBoxOpened(false);
+  }, []);
+
+  const gridBoxDisplayExpr = useCallback((item) => (
+    item && `${item.CompanyName} <${item.Phone}>`
+  ), []);
+
+  const treeViewOnContentReady = useCallback((e) => {
```

Comment:
Can be moved out of component

Reply:
uses treeboxvalue state inside, so it cant be moved